bitcoin/src/blockdata/witness.rs

```diff
@@ -390,6 +390,28 @@ impl Witness {
             .and_then(|script_pos_from_last| self.nth(len - script_pos_from_last))
             .map(Script::from_bytes)
     }
+
+    /// Get sigops for the Witness
+    ///
+    /// witness_version is the raw opcode. OP_0 is 0, OP_1 is 81, etc.
+    pub(crate) fn sig_ops(&self, prevout_script: &Script) -> usize {
+        let bytes = prevout_script.as_bytes();
+        match (bytes.first(), bytes.len()) {
+            // Segwit v0, P2WPKH
+            (Some(0), 22) => 1,
+            // Segwit v0, P2WSH
+            (Some(0), 34) => {
+                if let Some(n) =
+                    self.last().map(Script::from_bytes).map(|s| s.count_sigops_accurate())
+                {
+                    n
+                } else {
+                    0
+                }
+            }
+            _ => 0,
+        }
```

Comment:
I'm a bit brain fried so take this with a pinch of salt:
- `sig_ops` name is non-uniform with other functions, should be something like "count_sigops"
-  you can call `prevout_script.is_v0_p2wpkh()` and `is_v0_p2wsh()` instead of doing the first byte/bytes len matching.
- FWIW it kind of feels like this function is in the wrong module, I tried inlining it into `count_witness_sigops` but that was not that nice.

Reply:
Done, Done, and meh... it's not the best but inlining it with some better naming seemed the best choice.

Fixed.